tests/benchmarks/cli/commands/test_checkout.py

```diff
@@ -1,6 +1,38 @@
-def test_checkout(bench_dvc, tmp_dir, dvc, make_dataset):
-    dataset = make_dataset(dvcfile=True, cache=True, files=False)
-    bench_dvc("checkout", dataset)
-    bench_dvc("checkout", name="noop")
-    (dataset / "new").write_text("new")
-    bench_dvc("checkout", "--force", name="update")
+import os
+
+import pytest
+from dvc_objects.fs import generic
+from dvc_objects.fs.utils import tmp_fname
+
+from dvc.fs import localfs
+
+
+def _skip_unsupported_link(src, dest, link_type):
+    src_test_file = os.path.join(src, tmp_fname())
+    dest_test_file = os.path.join(dest, tmp_fname())
+    if not generic.test_links(
+        [link_type], localfs, src_test_file, localfs, dest_test_file
+    ):
+        pytest.skip(f"{link_type} not supported")
+
+
+def generate_test(*, link_type="copy"):
```

Comment:
I think we should try to support `pytest.parametrize`.

Reply:
@skshetry Maybe you could take a look at fixing that. I just didn't have the time to get into it yet.

Reply:
I’d prefer not to at this time. We can get back to it later. 

I’d even propose to just use `parametrize` and fix plotting/html generation later.

Reply:
One simple way would be to get rid of `dvc_revs` fixture and iterate on it manually outside. This removes the complexity of parsing and filtering revisions out.

Reply:
Let's not remove important features to fix one small bug 😄

Reply:
I'm just saying that `dvc-revs` could be implemented outside in a wrapper script. Why I am saying this is because we don't want to parametrize the test across revisions right, we want to run different _session_ of the same function across different revision.

It's a very small difference but it makes more sense in `pytest-benchmark` and `pytest` where each parametrized tests are different tests.

Reply:
I know it's convenient to have that, but I think it's okay to replace it with a wrapper as you locally don't really care much about `--dvc-revs`, it's more of a CI thing.